### PR TITLE
Added regular expressions into the tagging analyzer to provide more flexibility.

### PR DESCRIPTION
**Pull request by kiddinn**
This PR introduces the concept of a regular expression into the tagger analyzer.

+ Tagger analyzer now supports indicating a regular expression and an attribute to run the expression against
+ For each event that the original query finds a regular expression can be run in addition to the query to further narrow down if an event should be tagged or not.

**Checks**

- [ ] All tests succeed.
- [ ] Unit tests added.
- [ ] e2e tests added.
- [x] Documentation updated.

### CODE REVIEW

File: timesketch/lib/analyzers/tagger.py [hunk 1]
```diff
@@ -58,26 +64,62 @@ def tagger(self, name, config):
...
             event_counter += 1
             event.add_tags(tags)
             event.add_emojis(emojis_to_add)
 
             # Commit the event to the datastore.
             event.commit()
 
-        if create_view and event_counter:
+        if save_search and event_counter:
             self.sketch.add_view(
```
[the remark this answers is not in the document]
Comment (kiddinn) [reply timesketch/lib/analyzers/tagger.py:121]: you mean the `add_view`? I was thinking about that, that's a change in the interface and then a change in most of the analyzers, I think I'll do that in a follow up PR,

[the remark this answers is not in the document]
Comment (berggren) [reply timesketch/lib/analyzers/tagger.py:121]: Yeah, definitely do that in a follow up. 👍

File: data/tags.yaml [hunk 2]
```diff
@@ -2,46 +2,60 @@
...
-#        view_name: 'TEST view name'
+#        save_search: true
+#        search_name: 'TEST the tag'
+#
+# test_tagger_with_re:
+#        query_string: 'test'
+#        tags: ['test-retag']
+#        emojis: ['FISHING_POLE']
+#        save_search: true
+#        regular_expression: '[a-z][A-C]?://foobarX{0,4}'
```
Comment (berggren) [inline data/tags.yaml:31]: Do we want to add re_flags as well? In features.yaml we do support flags:
```
re: '([a-zA-Z0-9_\.+\-]+@[a-zA-Z0-9\-]+\.[a-zA-Z0-9\-\.]+)'
re_flags: []
```

File: timesketch/lib/analyzers/tagger.py [hunk 3]
```diff
@@ -58,26 +66,62 @@ def tagger(self, name, config):
...
+                attributes = [attribute]
+
         event_counter = 0
-        events = self.event_stream(query_string=query, query_dsl=query_dsl)
+        events = self.event_stream(
+            query_string=query, query_dsl=query_dsl, return_fields=attributes)
 
         for event in events:
+            if expression:
+                value = event.source.get(attributes[0])
```
Comment (berggren) [inline timesketch/lib/analyzers/tagger.py:107]: Whats the idea with attributes[0] here - do you plan to support multiple attributes in the future?

**Review by berggren — CHANGES_REQUESTED** [review]
Looks good - a couple of questions

File: data/tags.yaml (hunk 2 above)
Comment (kiddinn) [reply data/tags.yaml:31]: sure, SGTM, I'll add that

File: timesketch/lib/analyzers/tagger.py (hunk 3 above)
Comment (kiddinn) [reply timesketch/lib/analyzers/tagger.py:107]: yes, that is the idea, to keep that open, since we might do that

**Review by kiddinn — COMMENTED** [review]
done, PTAL

**Review by berggren — APPROVED** [review]

**Pull request merged**